### ISSUE TITLE
StatusCodeError: 400 with uploadSession

### ISSUE THREAD

**Issue by JonasGruenwald** [dkatavic/onedrive-api#16]
Hi, I'm trying to implement a file upload using this library, `uploadSimple` works great, but most of the files I want to upload will be over 10mb, and when I use `uploadSession` with the same parameters (and added fileSize), I get the following error:
```
Unhandled rejection StatusCodeError: 400 - [object Object]
    at new StatusCodeError (/Users/jonas/Projects/personal/pipeline/uploader/node_modules/request-promise/lib/errors.js:26:15)
    at Request.RP$callback [as _callback] (/Users/jonas/Projects/personal/pipeline/uploader/node_modules/request-promise/lib/rp.js:68:32)
    at Request.self.callback (/Users/jonas/Projects/personal/pipeline/uploader/node_modules/request/request.js:185:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/Users/jonas/Projects/personal/pipeline/uploader/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/Users/jonas/Projects/personal/pipeline/uploader/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
I haven't really used streams before so I'm not sure if I'm doing everything correctly, I do it as follows:

- I get the `stats` object for the file I want to upload and use `stats.size` to determine the fileSize
- I use `fs.createReadStream(localPath)` to create the readable stream

Do you have any idea what could be going wrong? The access token and path is for sure correct, since `uploadSimple` works just fine with the same parameters.
Here is the full parameter object I use to call `uploadSession`:

```
{ fileSize: 16805688,
  accessToken:'<my-token>',
  filename: 'test5.png',
  parentPath: '/pipeline/',
  readableStream:
   ReadStream {
     _readableState:
      ReadableState {
        objectMode: false,
        highWaterMark: 65536,
        buffer: BufferList { head: [Object], tail: [Object], length: 1 },
        length: 65536,
        pipes: null,
        pipesCount: 0,
        flowing: null,
        ended: false,
        endEmitted: false,
        reading: false,
        sync: false,
        needReadable: true,
        emittedReadable: true,
        readableListening: false,
        resumeScheduled: false,
        paused: true,
        emitClose: false,
        destroyed: false,
        defaultEncoding: 'utf8',
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: true,
     _events: [Object: null prototype] { end: [Function] },
     _eventsCount: 1,
     _maxListeners: undefined,
     path:
      '<absolute path to this image>/697DD.png',
     fd: 21,
     flags: 'r',
     mode: 438,
     start: undefined,
     end: Infinity,
     autoClose: true,
     pos: undefined,
     bytesRead: 65536,
     closed: false } }
```

**Comment by JonasGruenwald** [comment]
I just saw you mention an issue with uploading to a path in this PR https://github.com/dkatavic/onedrive-api/pull/15 so I tried to omit the parentPath parameter and it does work that way!
So I suppose the issue lies there, I can actually live without uploading to a specified path, so feel free to close this.

**Comment by dkatavic** [comment]
Hi @JonasGruenwald . Sorry for the late reply. 

Upload session with a parent path should work, so I will leave this issue open. I will try to get time later today to fix it and write test for it

**Comment by dkatavic** [comment]
Hi @JonasGruenwald , this is fixed now with #17 

Sorry for the bug. Behavior with `parentPath` is not properly documented in https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_createuploadsession?view=odsp-graph-online so implementation took a bit of guessing. I have written an e2e test for that behavior so it should be working fine now. Let me know if there are any other issues

**Comment by JonasGruenwald** [comment]
Hi @dkatavic, that's super cool! Will definitely be convenient to have that option.
No need to apologize, thanks for providing this great package, makes working with OneDrive a lot easier. 👍
